tests/analyze_assoc.R

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/Rscript
+#
+# Copyright 2015 Google Inc. All rights reserved.
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+library("getopt")
+source("../analysis/R/encode.R")
+source("../analysis/R/decode.R")
+source("../analysis/R/simulation.R")
+source("../analysis/R/read_input.R")
+source("../analysis/R/association.R")
+
+# This function caches the .csv as an .rda for faster loading.  NOTE: It
+# assumes the map csv file is immutable. This isn't true if you
+# re-run assoc_sim.R. Adjust immutable_flag as required.
+# Modified from analysis/R/read_input.R to load 2 or more map files
+# into the environment
```

Comment:
Is the format exactly the same, and it's just loading 2 map files?  In that case it should definitely be factored out to reuse the logic and not copy and paste.  We already have a use case for 3 or more variables anyway.  Even if the analysis doesn't support it yet, there's no reason for this plumbing not to.

I also don't like the use of globals, but that was already there.  I think there is a better way, which I will experiment with.

And I think it should go in read_input.R so we have all the file formats in one place.  You're already depending on that file anyway.